src/components/organisms/NewResourceWizard/NewResourceWizard.tsx

```diff
@@ -230,6 +303,16 @@ const NewResourceWizard = () => {
         form.setFieldsValue({selectedResourceId: SELECT_OPTION_NONE});
       }
     }
+
+    if (savingDestination === 'saveToFolder') {
+      const generatedFileName = await generateFullFileName(
```

Comment:
This should be done in an `useEffect` , everytime the `selectedFolder` variable is changed

Reply:
I have moved the logic to useEffect now